caps/repo.go

```diff
@@ -71,8 +71,8 @@ func (r *Repository) Add(cap *Capability) error {
 	return nil
 }
 
-// HasType checks if the repository contains the given type.
-func (r *Repository) HasType(t *Type) bool {
+// hasType is identical to HasType but doesn't hold the repository lock
```

Comment:
And you expect we, wannabe robots, to never mistype h by H or the opposite on every context? :-)

This should be named unlockedHasType or something. Or, alternatively, can we get rid of the public version altogether? I can't see many practical public use cases for a full-type comparison like this.